### ISSUE TITLE
support for bash-like expansion modifiers

### ISSUE THREAD

**Issue by rsgalloway** [rsgalloway/envstack#12]
Add support for bash-like parameter expansion and expansion modifier syntax. 

| Feature            | Description |
|---------------------|-------------|
| ${VAR:-default}  |  os.environ.get('VAR', 'default') |
| ${VAR:=default}  |  VAR = VAR or 'default' |
| ${VAR:?error message} |  if not VAR: raise ValueError() |

Optional:

| Feature            | Description |
|---------------------|-------------|
| ${VAR/pattern/replacement} | var.replace('pattern', 'replacement') |
| ${VAR^^} | VAR.upper() |
| ${VAR,,} | VAR.lower() |

**Comment by rsgalloway** [comment]
with this changed, values cannot be overridden unless using one of the default value modifiers:


```yaml
all: &default
  ENV: prod
  HELLO: ${HELLO:=world}
```

with default value modifier:
```bash
$ envstack -- echo {HELLO}
world
$ HELLO=goodbye envstack -- echo {HELLO}
goodbye
```

without default value modifier:
```bash
$ envstack -- echo {ENV}
prod
$ ENV=dev eevnvstack -- echo {ENV}
prod
```

**Comment by rsgalloway** [comment]
released in version 0.7.0
